doc/release-notes-31384.md

```diff
@@ -0,0 +1,34 @@
+- Node and Mining
+
+---
+
+- **PR #31384** fixed an issue where block reserved weight for fixed-size block header, transactions count,
+  and coinbase transaction was done in two separate places.
+  Before this pull request, the policy default for the maximum block weight was `3,996,000` WU, calculated by
+  subtracting `4,000 WU` from the `4,000,000 WU` consensus limit to account for the fixed-size block header,
+  transactions count, and coinbase transaction. During block assembly, Bitcoin Core clamped custom `-blockmaxweight`
+  value to not be more than the policy default.
+
+  Additionally, the mining code added another `4,000 WU` to the initial reservation, reducing the effective block template
+  size to `3,992,000 WU`.
+
+  Due to this issue, the total reserved weight was always `8,000 WU`, meaning that even when specifying a `-blockmaxweight`
+  higher than the policy default, the actual block size never exceeded `3,992,000 WU`.
+
+  The fix consolidates the reservation into a single place and introduces a new startup option,
+  `-blockreservedweight` (default: `8,000 WU`). This startup option specifies the reserved weight for
+  the fixed-size block header, transactions count, and coinbase transaction.
+  The default value of `-blockreservedweight` was chosen to preserve the previous behavior.
+
+  **Upgrade Note:** The default `-blockreservedweight` ensures backward compatibility for users who relied on the previous behavior.
+
+  Users who manually set `-blockmaxweight` to its maximum value of `4,000,000 WU` should be aware that this
+  value previously had no effect since it was clamped to `3,996,000 WU`.
+
+  Users lowering `-blockreservedweight` should ensure that the total weight (for the block header, transaction count, and coinbase transaction)
+   does not exceed the reduced value.
+
+  As a safety check, Bitcoin core will **fail to start** when `-blockreservedweight` init parameter value is lower than `2000` weight units.
+
+  Bitcoin Core will also **fail to start** if the `-blockmaxweight` or `-blockreservedweight` init parameter exceeds
```

Comment:
Maybe also clearly state how they interact? What if you set `-blockreservedweight` higher than `-blockmaxweight`?

Reply:
That's an absurd thing to do, so I'm worried that explaining it just makes the text hard to follow. That could miners to ignore the import warning about setting this value too low.

Reply:
Or have a test for the expected behavior? A comment in the options-handling code? I don't think it should be unspecified.

Reply:
There are two cases to this.

1. When the `-blockmaxweight` is default (`4,000,000 WU`), and you set a higher block reserved weight, the node will fail to start, there is a test for this in this PR


2. When `-blockmaxweight` is custom i.e lower that (`4,000,000 WU`) and you set a block reserved weight to a value
       a) > `4,000,000WU` the node will fail to start (There is a test for this)
        b) < `4000,000 WU` the node will start but all generated block template will be empty when the value is higher than the custom`-blockmaxweight` value. ( No test for this, and it's an absurd behavior as @Sjors mentioned I didn't add the test because we would want to removed `-blockmaxweight` in the future since we now have `-blockreservedweight`

Reply:
It's explained in `BlockAssembler::Options ClampOptions`, though we should probably point to that from elsewhere.

Reply:
Also see https://github.com/bitcoin/bitcoin/pull/31384#discussion_r1937951204

Reply:
(our comments came in at the same time)

Reply:
Ok, makes sense to get rid of `-blockmaxweight`. What do you think the plan for that should be? If I understand you correctly that users should really only use one of them, should we be deprecating it at the same time as introducing `-blockreservedweight`?

Reply:
I think we should deprecate it in next release and then remove it later.

Because ocean mining currently heavily rely on `-blockmaxweight`